pkg/auth/cluster/cluster.go

```diff
@@ -77,11 +77,10 @@ func verifySource(ctx context.Context, validKeys [][]byte) error {
 }
 
 // Authorized returns whether the context has been identified as a cluster call.
-// It panics if it does not inherit from `NewContext`.
 func Authorized(ctx context.Context) error {
 	ok, isStored := ctx.Value(clusterAuthKey).(bool)
 	if !isStored {
-		panic("call source not verified; did you register the cluster authentication hook for this call?")
+		return errNoClusterKey.New()
```

Comment:
This panic should be restored. It was used to indicate that the implementation of an RPC checks for cluster auth, but the hook that injects cluster auth, is not executed in the chain.

See also

https://github.com/TheThingsNetwork/lorawan-stack/blob/3fac99e1a10caad41238b8bf159da19942207b3f/pkg/identityserver/identityserver.go#L152-L175

Reply:
I know what it was used for, but I have a problem with this, namely that we cannot check if cluster auth was used.

Panics are for developer mistakes, but I don't see what's the mistake is. We have places in our service layer where different, entirely unrelated authentication types can be used: TLS client auth, entity access rights and cluster auth.

So I disagree with panicking, because those are for developer mistakes and this isn't any, but you probably have a better reason that I'm not aware of, so I will make the authentication depending on the origin.